alicloud/data_source_alicloud_kms_ciphertext.go

```diff
@@ -0,0 +1,68 @@
+package alicloud
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/kms"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func dataSourceAlicloudKmsCiphertext() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudKmsCiphertextRead,
+
+		Schema: map[string]*schema.Schema{
+			"plaintext": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"context": {
```

Comment:
I think use `encryption_context ` will be better. 
In addition, it should be `TypeMap`.

Reply:
Agreed, I changed the parameter accordingly.